modules/metastorage/src/main/java/org/apache/ignite/metastorage/internal/MetaStorageManager.java

```diff
@@ -75,77 +96,195 @@ public MetaStorageManager(
+    }
+
+    /**
+     * @see MetaStorageService#putAll(Map)
+     */
+    public @NotNull CompletableFuture<Void> putAll(@NotNull Map<Key, byte[]> vals) {
+        return svc.putAll(vals);
+    }
 
-    public void registerWatch() {
+    /**
+     * @see MetaStorageService#getAndPutAll(Map)
+     */
+    public @NotNull CompletableFuture<Map<Key, Entry>> getAndPutAll(@NotNull Map<Key, byte[]> vals) {
+        return svc.getAndPutAll(vals);
+    }
 
+    /**
+     * @see MetaStorageService#remove(Key)
+     */
+    public @NotNull CompletableFuture<Void> remove(@NotNull Key key) {
+        return svc.remove(key);
     }
 
-    public void deployWatches() {
+    /**
+     * @see MetaStorageService#getAndRemove(Key)
+     */
+    public @NotNull CompletableFuture<Entry> getAndRemove(@NotNull Key key) {
+        return svc.getAndRemove(key);
+    }
+
+    /**
+     * @see MetaStorageService#removeAll(Collection)
+     */
+    public @NotNull CompletableFuture<Void> removeAll(@NotNull Collection<Key> keys) {
+        return svc.removeAll(keys);
+    }
+
+    /**
+     * @see MetaStorageService#getAndRemoveAll(Collection)
+     */
+    public @NotNull CompletableFuture<Map<Key, Entry>> getAndRemoveAll(@NotNull Collection<Key> keys) {
+        return svc.getAndRemoveAll(keys);
+    }
+
+    /**
+     * @see MetaStorageService#invoke(Key, Condition, Operation, Operation)
+     */
+    public @NotNull CompletableFuture<Boolean> invoke(@NotNull Key key, @NotNull Condition cond,
+        @NotNull Operation success, @NotNull Operation failure) {
+        return svc.invoke(key, cond, success, failure);
+    }
+
+    /**
+     * @see MetaStorageService#getAndInvoke(Key, Condition, Operation, Operation)
+     */
+    public @NotNull CompletableFuture<Entry> getAndInvoke(@NotNull Key key, @NotNull Condition cond,
+        @NotNull Operation success, @NotNull Operation failure) {
+        return svc.getAndInvoke(key, cond, success, failure);
+    }
+
+    /**
+     * @see MetaStorageService#range(Key, Key, long)
+     */
+    public @NotNull Cursor<Entry> range(@NotNull Key keyFrom, @Nullable Key keyTo, long revUpperBound) {
+        return svc.range(keyFrom, keyTo, revUpperBound);
+    }
+
+    /**
+     * @see MetaStorageService#range(Key, Key)
+     */
+    public @NotNull Cursor<Entry> range(@NotNull Key keyFrom, @Nullable Key keyTo) {
+        return svc.range(keyFrom, keyTo);
+    }
+
+    /**
+     * @see MetaStorageService#compact()
+     */
+    public @NotNull CompletableFuture<Void> compact() {
+        return svc.compact();
+    }
+
+    /**
+     * Stop current batch of consolidated watches and register new one from current {@link WatchAggregator}.
+     *
+     * @return Ignite UUID of new consolidated watch
+     */
+    private synchronized CompletableFuture<IgniteUuid> updateWatches() {
+        var range = watchAggregator.inferGeneralCriteria();
+        igniteUuid = igniteUuid
+            .thenCompose(svc::stopWatch)
```

Comment:
We can avoid redeploying the watch if the watchAggregator does not change.
Need to add a flag changed for the purpose.
But it might not need until this method to use from private API.

Reply:
The logic of redeploying will be updated during https://issues.apache.org/jira/browse/IGNITE-14604 and I will fix this issue too.